src/Clinic.DAL/Repositories/OfficeRepository.cs

```diff
@@ -1,43 +1,35 @@
 ﻿using Clinic.DAL.Entities;
 using Clinic.DAL.Interfaces;
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using static Clinic.DAL.Entities.EnumsEntity;
 
 namespace Clinic.DAL.Repositories
 {
-    public class OfficeRepository : IOfficeRepository
+    public class OfficeRepository : Repository<OfficeEntity>, IOfficeRepository
     {
-        public Task<OfficeEntity> AddAsync(OfficeEntity entity, CancellationToken cancellationToken)
-        {
-            throw new NotImplementedException();
-        }
+        public OfficeRepository(ClinicDbContext dbContext) : base(dbContext) { }
 
-        public Task<IEnumerable<OfficeEntity>> GetAllAsync(CancellationToken cancellationToken)
+        public async Task<IEnumerable<OfficeEntity>> GetAllAsync(string? address, string? phoneNumber, OfficeStatus? isActive, CancellationToken cancellationToken)
         {
-            throw new NotImplementedException();
-        }
+            IQueryable<OfficeEntity> query = _dbContext.Offices.AsQueryable();
 
-        public Task<IEnumerable<OfficeEntity>> GetAllAsync(string address, string phoneNumber, string isActive, CancellationToken cancellationToken)
-        {
-            throw new NotImplementedException();
-        }
+            if (!string.IsNullOrEmpty(address))
+            {
+                var firstNameLower = address.ToLower();
+                query = query.Where(x => EF.Functions.Like(x.Address.ToLower(), $"%{address}%"));
+            }
 
-        public Task<OfficeEntity> GetByIdAsync(Guid id, CancellationToken cancellationToken)
-        {
-            throw new NotImplementedException();
-        }
+            if (!string.IsNullOrEmpty(phoneNumber))
```

Comment:
Instead of IsNullOrEmpty it's much better to use IsNullOrWhiteSpace

https://stackoverflow.com/questions/18710644/difference-between-isnullorempty-and-isnullorwhitespace-in-c-sharp

Reply:
Is it only for phoneNumber or should be used for address too?